ParseCAN/data/log.py

```diff
@@ -1,12 +1,48 @@
 from pathlib import Path
-
+import ParseCAN as pcn
 
 class Log:
-    def unpack(self, spec, **kwargs):
-        return (spec.unpack(msg, **kwargs) for msg in self)
+    def unpack(self, spec, include_raw=False, **kwargs):
+        # TODO: add multiplex support
+
+        # Precompute which frame and bus each frame_id belongs to
+        # Top level frame ids are unique
+        frame_map = {}
+        bus_map = {}
+        bus_unique = spec.protocol['name']['can'].bus
+        for bus_nm in bus_unique['name']:
```

Comment:
This is not how you're supposed to do this. The idea is that you call unpack on the appropriate `Bus` and it figures out the rest. Essentially you are copying the entire spec in a different representation.